File: JAVA/025_1000_digit_Fibonacci_Number.java
import java.math.BigDecimal;

class TestMain 
{

    
	public static void main(String[] args) 
	{
		System.out.println(findRank(1000));

	}
	
	private static int findRank(int noOfDigits)
	{
		int counter=2;
		BigDecimal n1 = new BigDecimal(1);
		BigDecimal n2 = new BigDecimal(1);
		BigDecimal sum = new BigDecimal(0);
		while(sum.toString().length()<noOfDigits)
		{
			sum = n1.add(n2);
			n1 = n2;
			n2 = sum;
			counter++;
		}
		return counter;
	}

}

File: CPP/025_1000_digit_Fibonacci_Number.cpp
#include <iostream>
 
using namespace std;
 
int findRank(int noOfDigits);
 
        int main()
        {
                cout << findRank(1000) << endl;
        }
        
        int findRank(int noOfDigits)
        {
                short sum[noOfDigits],num1[noOfDigits],num2[noOfDigits];
                int carry = 0,number,result,i,counter=2;
                for(i=0;i<noOfDigits;i++)
                {
                        sum[i] = 0;
                        num1[i] = 0;
                        num2[i] = 0;
                }
                num1[i-1]=1;
                num2[i-1]=1;
                
                while(sum[0]==0)
                {
                    carry=0;
                    for(i=noOfDigits-1;i>=0;i--)
                    {
                        number=num1[i]+num2[i];
                        result = (number%10)+carry;
                        carry=number/10;
                        sum[i]=(short)result;
                    }
                    for(i=noOfDigits-1;i>=0;i--)
                    {
                        num1[i]=num2[i];
                        num2[i]=sum[i];
                    }
                    counter++;
                }
                return counter;
        }